### ISSUE TITLE
Benchmark against deno.land/x/mysql

### ISSUE THREAD

**Issue by sant123** [sail-sail/mysql2#1]
Hi, I have been working on some benchmarks that I want to share with you:

```ts
import { Client, configLogger } from "https://deno.land/x/mysql@v2.10.2/mod.ts";
import * as mysql2 from "https://deno.land/x/mysql2@v1.0.1/mod.ts";

await configLogger({ enable: false });

Deno.bench({
  name: "mysql",
  group: "database",
  baseline: true,
  fn: async () => {
    const mysqlClient = await new Client().connect({
      hostname: "localhost",
      username: "root",
      password: "your-root-password",
      db: "Jupiter",
      debug: false,
    });

    await mysqlClient.query("SELECT name, age FROM People");
    await mysqlClient.close();
  },
});

Deno.bench({
  name: "mysql2",
  group: "database",
  fn: async () => {
    const mysql2Pool = mysql2.createPool({
      host: "localhost",
      user: "root",
      password: "your-root-password",
      database: "Jupiter",
      connectionLimit: 4,
    });

    await mysql2Pool.query("SELECT name, age FROM People");
    await mysql2Pool.end();
  },
});

```

The results are:

![image](https://user-images.githubusercontent.com/7959437/171323964-98889e4e-26c9-4e0e-999d-bec6ac629206.png)

However we can preserve our created pool and change a little the benchmark.

```ts
import { Client, configLogger } from "https://deno.land/x/mysql@v2.10.2/mod.ts";
import * as mysql2 from "https://deno.land/x/mysql2@v1.0.1/mod.ts";

await configLogger({ enable: false });

const mysqlClient = await new Client().connect({
  hostname: "localhost",
  username: "root",
  password: "your-root-password",
  db: "Jupiter",
  debug: false,
});

const mysql2Pool = mysql2.createPool({
  host: "localhost",
  user: "root",
  password: "your-root-password",
  database: "Jupiter",
  connectionLimit: 4,
});

Deno.bench({
  name: "mysql",
  group: "database",
  baseline: true,
  fn: async () => {
    await mysqlClient.query("SELECT name, age FROM People");
  },
});

Deno.bench({
  name: "mysql2",
  group: "database",
  fn: async () => {
    await mysql2Pool.query("SELECT name, age FROM People");
  },
});
```

The results change:

![image](https://user-images.githubusercontent.com/7959437/171324192-8f95cfeb-3321-4652-8fdf-8d09517937e1.png)

So basically I can conclude that the pool creation is taking some time to initialize. Do you think that can be improved? Just really curious about it.

Furthermore thank you for this library!

**Comment by sail-sail** [comment]
Thank you!
In consideration of the pool initialize in application only once, so almost no impact

**Comment by sant123** [comment]
> Thank you!
> 
> In consideration of the pool initialize in application only once, so almost no impact

Makes sense, after creating the pool the queries performance is really good. 

Closing.
